File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

void grade(string text)
{
    int len_text = strlen(text);

    float length=0;
    for(int i=0; i<len_text; i++)
    {
        if((text[i]>= 'a' && text[i]<= 'z') || (text[i]>= 'A' && text[i]<= 'Z'))
            length +=1;
    }

    float sentence=0;
    for(int i=0; i<len_text; i++)
    {
        if((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
            sentence +=1;
    }

    float W=1;
    for(int i=0; i<len_text; i++)
    {
        if(text[i] == ' ')
            W +=1;
    }

    float L = (length*100)/W;
    float S = (sentence*100)/W;

    int index=round((0.0588*L)-(0.296*S)-(15.8));

    if (index>= 16)
        printf("Grade 16+\n");
    else if (index<1)
        printf("Before Grade 1\n");
    else
        printf("Grade %i\n", index);
}


int main(void)
{
    string text = get_string("Text: ");
    grade(text);
}

File: pset6/readability/readability.py
from cs50 import get_string
import re

def main():
    sentence = get_string("Text: ")
    result = grade(sentence)
    if (result >= 16):
        print("Grade 16+")
    elif (result < 1):
        print("Before Grade 1")
    else:
        print("Grade ", result)


def grade(sentence):
    count_s =0
    count_w =1
    count_l =0
    for i in range(len(sentence)):
        if((sentence[i] == '!') or (sentence[i] == '.') or (sentence[i] == '?')):
            count_s +=1
        elif (sentence[i]== " "):
            count_w +=1
        elif(sentence[i].isalpha()):
            count_l +=1

    L = (count_l * 100.0)/count_w
    S = (count_s * 100.0)/count_w
    index = round( (0.0588 * L) - (0.296 * S) - 15.8)
    return index


main()

